### PR TITLE
dlfcn: add RTLD_DEFAULT macro value

### LINKED ISSUE #134: Give the ability to perform dlopen(NULL)
As per the man page, `dlopen(NULL)` is supposed to return the equivalent of the macro `RTLD_DEFAULT` which not accessible from go code obviously and is implementation defined. This function is especially useful to run already loaded symbols like the ones loaded with the pragma `go:cgo_import_dynamic`. 

But since we pass a string to Dlopen, a nil value cannot be passed as parameter. How could we give access to this feature ?

### PR DESCRIPTION
**Pull request by eliottness**
Closes #134

### CODE REVIEW

File: dlfcn_test.go [hunk 1]
```diff
@@ -14,10 +14,18 @@ import (
 	"runtime"
 	"strings"
 	"testing"
+	_ "unsafe" // for go:linkname
```
Comment (TotallyGamerJet) [inline dlfcn_test.go:17]: I don't see the go linkname

**Review by TotallyGamerJet — CHANGES_REQUESTED** [review]

File: dlfcn_test.go (hunk 1 above)
Comment (eliottness) [reply dlfcn_test.go:17]: It's an artifact from a previous version, removing it
```suggestion
```

**Review by TotallyGamerJet — COMMENTED** [review]
LGTM! @hajimehoshi, do you approve the API addition?

File: dlfcn_test.go [hunk 2]
```diff
@@ -18,6 +18,13 @@ import (
 	"github.com/ebitengine/purego"
 )
 
+func TestSimpleDlsym(t *testing.T) {
+	_, err := purego.Dlsym(purego.RTLD_DEFAULT, "dlsym")
+	if err != nil {
```
Comment (hajimehoshi) [inline dlfcn_test.go:23]:
```go
if _, err := purego.Dlsym(purego.RTLD_DEFAULT, "dlsym"); err != nil {
```

File: dlfcn_test.go [hunk 3]
```diff
@@ -18,6 +18,13 @@ import (
...
+		t.Errorf("failed to dlsym: %s", err)
```
Comment (hajimehoshi) [inline dlfcn_test.go:24]: `Dlsym with RTLD_DEFAULT failed: %v`

File: dlfcn_test.go [hunk 4]
```diff
@@ -18,6 +18,12 @@ import (
 	"github.com/ebitengine/purego"
 )
 
+func TestSimpleDlsym(t *testing.T) {
+	if _, err := purego.Dlsym(purego.RTLD_DEFAULT, "dlsym"); err != nil {
+		t.Errorf("Dlsym with RTLD_DEFAULT failed: %s", err)
```
Comment (hajimehoshi) [inline dlfcn_test.go:23]: %v, not %s

**Review by hajimehoshi — APPROVED** [review]
LGTM

**Review by TotallyGamerJet — APPROVED** [review]

**Pull request merged**